aioitertools/builtins.py

```diff
@@ -13,22 +13,41 @@
 
 import asyncio
 import builtins
+import inspect
 from typing import (
     Any,
     AsyncIterable,
     AsyncIterator,
+    Awaitable,
     Callable,
     Iterable,
     List,
     Set,
     Tuple,
+    Union,
     cast,
     overload,
 )
 
 from .types import AnyIterable, AnyIterator, AnyStop, T, R, T1, T2, T3, T4, T5
 
 
+@overload
+async def maybe_await(object: Awaitable[T]) -> T:
+    pass
+
+
+@overload
+async def maybe_await(object: T) -> T:
+    pass
+
+
+async def maybe_await(object: Union[Awaitable[T], T]) -> T:
+    if inspect.isawaitable(object):
+        return await object  # type: ignore
+    return object  # type: ignore
```

Comment:
Might make sense to put this in a `helpers` module and import from there, so that we better differentiate between what is shadowing the system builtins and what's not.  Probably couldn't hurt to also define `__all__`, but no need for that in this diff.